src/main/java/com/connexta/store/service/impl/StoreServiceImpl.java

```diff
@@ -6,128 +6,208 @@
  */
 package com.connexta.store.service.impl;
 
-import com.connexta.store.adaptors.FileRetrieveResponse;
 import com.connexta.store.adaptors.StorageAdaptor;
 import com.connexta.store.adaptors.StorageAdaptorRetrieveResponse;
 import com.connexta.store.clients.IndexDatasetClient;
 import com.connexta.store.clients.TransformClient;
 import com.connexta.store.controllers.StoreController;
-import com.connexta.store.exceptions.IndexDatasetException;
 import com.connexta.store.exceptions.RetrieveException;
 import com.connexta.store.exceptions.StoreException;
-import com.connexta.store.exceptions.TransformException;
+import com.connexta.store.exceptions.common.DetailedResponseStatusException;
+import com.connexta.store.poller.TransformStatusTask;
+import com.connexta.store.rest.models.MetadataInfo;
+import com.connexta.store.service.api.IonData;
 import com.connexta.store.service.api.StoreService;
+import java.io.IOException;
 import java.io.InputStream;
+import java.net.MalformedURLException;
 import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.util.List;
 import java.util.Map;
 import java.util.UUID;
-import javax.validation.constraints.Max;
-import javax.validation.constraints.Min;
+import java.util.concurrent.BlockingQueue;
 import javax.validation.constraints.NotBlank;
 import javax.validation.constraints.NotNull;
-import javax.validation.constraints.Pattern;
-import javax.validation.constraints.Size;
 import lombok.AllArgsConstructor;
 import lombok.extern.slf4j.Slf4j;
+import org.springframework.core.io.Resource;
```

Comment:
I think we should avoid Spring-specific code in our business logic. The response should be handled in the controller.

Reply:
Yeah we are going to have to come in here and rip out all the spring code after we re-define our interfaces.